src/utils/simplifier.js

```diff
@@ -154,8 +154,8 @@ function simplify(realm, value: Value, isCondition: boolean = false): Value {
       if (Path.implies(notc)) return y;
       if (!isCondition) {
         if (Path.implies(AbstractValue.createFromBinaryOp(realm, "===", value, x))) return x;
-        if (Path.implies(AbstractValue.createFromBinaryOp(realm, "!==", value, x))) return y;
-        if (Path.implies(AbstractValue.createFromBinaryOp(realm, "!==", value, y))) return x;
+        if (!x.mightBeNumber() && Path.implies(AbstractValue.createFromBinaryOp(realm, "!==", value, x))) return y;
```

Comment:
Maybe a more concrete function, like x.mightBeNaN?

Reply:
Good idea, but probably something to do as a separate PR that fixes all overly general uses of mightBeNumber. Also, I can't really test the distinction until we have range domains.